Description: Support gpg_signing_key=default to indicate using the default GPG key.
Origin: commit, revision id: jelmer@jelmer.uk-20190109020153-kmu86u2fa0sfsegm
Author: Jelmer VernooÄ³ <jelmer@jelmer.uk>
Last-Update: 2019-01-09
Applied-Upstream: no
X-Bzr-Revision-Id: jelmer@jelmer.uk-20190109020153-kmu86u2fa0sfsegm

=== modified file 'breezy/gpg.py'
--- old/breezy/gpg.py	2018-11-19 01:38:53 +0000
+++ new/breezy/gpg.py	2019-01-09 02:01:53 +0000
@@ -200,14 +200,18 @@
     def _get_signing_keys(self):
         import gpg
         keyname = self._config_stack.get('gpg_signing_key')
+        if keyname == 'default':
+            # Leave things to gpg
+            return []
+
         if keyname:
             try:
                 return [self.context.get_key(keyname)]
             except gpg.errors.KeyNotFound:
                 pass
 
-        if keyname is None or keyname == 'default':
-            # 'default' or not setting gpg_signing_key at all means we should
+        if keyname is None:
+            # not setting gpg_signing_key at all means we should
             # use the user email address
             keyname = config.extract_email_address(
                 self._config_stack.get('email'))

